resolwe/flow/executors/docker/prepare.py

```diff
@@ -37,10 +37,7 @@ def resolve_data_path(self, data=None, filename=None):
         if data is None:
             return constants.DATA_ALL_VOLUME
 
-        if filename is None:
-            return os.path.join(constants.DATA_ALL_VOLUME, str(data.id))
-
-        return os.path.join(constants.DATA_ALL_VOLUME, str(data.id), filename)
+        return data.location.get_path(prefix=constants.DATA_ALL_VOLUME, filename=filename)
```

Comment:
Done, but it might be confusing because a few lines above you have 

```py
if not getattr(settings, 'FLOW_DOCKER_DONT_PULL', False):
```

where `settings` are `from django.conf import settings`. Might be worth adding a comment there too.